contracts/modules/v2/Assessment.sol

```diff
@@ -35,238 +27,286 @@ contract Assessment is IAssessment, MasterAwareV2 {
-    config.rewardRatio = 52; // 0.52%
-    config.incidentExpectedPayoutRatio = 3000; // 30%
-    config.claimAssessmentDepositRatio = 500; // 5% i.e. 0.05 ETH submission flat fee
+  constructor(address nxmAddress) {
+    nxm = INXMToken(nxmAddress);
+  }
+
+  function initialize (address masterAddress) external {
     config.minVotingPeriodDays = 3; // days
-    config.maxVotingPeriodDays = 30; // days
     config.payoutCooldownDays = 1; //days
     master = INXMMaster(masterAddress);
-    nxm = INXMToken(master.tokenAddress());
   }
 
+
   /* ========== VIEWS ========== */
 
-  function getVoteCountOfAssessor(address assessor) external override view returns (uint) {
-    return votesOf[assessor].length;
+  function min(uint a, uint b) internal pure returns (uint) {
+    return a <= b ? a : b;
   }
 
-  function getClaimsCount() external override view returns (uint) {
-    return claims.length;
+  function getVoteCountOfAssessor(address assessor) external override view returns (uint) {
+    return votesOf[assessor].length;
   }
 
-  function getIncidentsCount() external override view returns (uint) {
-    return incidents.length;
+  function getAssessmentsCount() external override view returns (uint) {
+    return assessments.length;
   }
 
   /* === MUTATIVE FUNCTIONS ==== */
 
-  /**
-   *  Submits a claim for assessment
-   *
-   *  @dev This function requires an ETH submission fee. See: _getSubmissionFee()
-   *
-   *  @param coverId          Cover identifier
-   *  @param requestedAmount  The amount expected to be received at payout
-   *  @param hasProof         When true, a ProofSubmitted event is emitted with ipfsProofHash.
-   *                          When false, no ProofSubmitted event is emitted to save gas if the
-   *                          cover wording doesn't enforce a proof of loss.
-   *  @param ipfsProofHash    The IPFS hash required for proof of loss. It is ignored if hasProof
-   *                          is false
-   */
-  function submitClaim(
-    uint24 coverId,
-    uint96 requestedAmount,
-    bool hasProof ,
-    string calldata ipfsProofHash
-  ) external payable override onlyMember {
-    AssessmentClaimsLib.submitClaim(
-      config,
-      internalContracts,
-      claims,
-      claimants,
-      coverId,
-      requestedAmount,
-      hasProof ,
-      ipfsProofHash
-    );
-
-  }
-
-  function submitIncident(
-    uint24 productId,
-    uint96 priceBefore,
-    uint32 date
-  ) external override {
-    (
-      AffectedToken memory affectedToken,
-      Incident memory incident
-    ) = AssessmentIncidentsLib.getIncidentToSubmit(
-      config,
-      IMemberRoles(getInternalContractAddress(ID.MR)),
-      productId,
-      priceBefore,
-      date
-    );
-
-    AssessmentIncidentsLib.saveIncident(
-      incident,
-      incidents,
-      affectedToken,
-      tokenAffectedByIncident
-    );
-  }
-
   function depositStake(uint96 amount) external override onlyMember {
```

Comment:
does it make sense to revert if the given amount is 0?

Reply:
I'd go for 
`stake` instead of `depositStake` and 
`unstake` instead of `withdrawStake`

Reply:
User handholding of sending zero amounts should be offloaded to the UI as long as there's no security benefits on the business side. Checks like this add up to the gas cost and the deployed bytecode size. Check out approve/transfer from OZ's ERC20, they allow 0 amounts.